playbooks/roles/cloud_host/tasks/main.yaml

```diff
@@ -1,4 +1,22 @@
 ---
+- name: "Set cloud env"
+  set_fact:
+    cloud: "{{ hostvars[host].location.cloud }}"
+
+- name: "Read vault data"
+  no_log: true
+  community.hashi_vault.vault_read:
+    url: "{{ ansible_hashi_vault_addr }}"
+    token: "{{ ansible_hashi_vault_token }}"
+    path: "{{ clouds[cloud].vault_path }}"
+  register: "cloud_token"
+  when: "clouds[cloud].vault_path is defined"
+
+- name: "Set cloud_auth variable with auth artifacts from vault"
+  no_log: true
+  set_fact:
+    cloud_auth: "{{ cloud_token.data.data }}"
```

Comment:
set_fact is bad cause it is stored in the facts cache. If this is really required (and here it is not really required) we can drop it and do cloud_token.data.data every time we need it. Alternatively we may want to unset fact at the end. This is not critical for us here, but generally we should really keep this in mind

Reply:
fixed